### DISCUSSION TITLE
How to connect helix to a *running* language server?

### DISCUSSION THREAD

**Discussion by thomasaarholt** [helix-editor/helix#7744] [Q&A] [answered]
I'm writing a SQL language server and would like to be able to monitor and debug the LSP logs directly. How can I prevent helix from starting up a new instance of the LSP server each time, and instead connect to the one I have running in a terminal instance?

**Comment by pascalkuthe** [comment, marked as answer]
That's not possible, the LSP is usually a single client multi server paradigm. The editor spawns and fully owns the server, communication is done over stdout/stdin pipes. If you start your server externally helix can not connect to stdout/stdin pipes.


You can simply test the server by running helix with `hx -v` which will print any output of the server into hekixes log file (which you can show with tail -f)

**Reply by thomasaarholt** [reply]
Thanks. I guess I can just log all my LSP logs to a single file as well, and refresh that file in a text editor on each test run. Thanks!
